src/server/service/diagnostics.rs

```diff
@@ -179,46 +178,55 @@ mod sys {
             item.set_pairs(pairs.into());
             collector.push(item);
         }
-        // https://www.kernel.org/doc/Documentation/filesystems/proc.txt
-        // We ignore the error because some operating systems don't contain procfs.
-        if let Ok(file) = File::open("/proc/stat") {
-            let names = vec![
-                "user",
-                "nice",
-                "system",
-                "idle",
-                "iowait",
-                "irq",
-                "softirq",
-                "steal",
-                "guest",
-                "guest_nice",
-            ];
-            for line in BufReader::new(file).lines() {
-                if let Ok(line) = line {
-                    if !line.starts_with("cpu") {
-                        continue;
-                    }
-                    let mut parts = line.split_whitespace();
-                    let name = match parts.nth(0) {
-                        Some(name) if name != "cpu" => name,
-                        _ => continue,
-                    };
-                    let mut pairs = vec![];
-                    for (val, name) in parts.zip(&names) {
-                        let mut pair = ServerInfoPair::default();
-                        pair.set_key((*name).to_string());
-                        pair.set_value(val.to_string());
-                        pairs.push(pair);
-                    }
-                    let mut item = ServerInfoItem::default();
-                    item.set_tp("cpu".to_string());
-                    item.set_name(name.to_string());
-                    item.set_pairs(pairs.into());
-                    collector.push(item);
-                }
-            }
+
+        let t1 = LiunxStyleCpuTime::current();
+        if t1.is_err() {
+            return;
+        }
+        let t1 = t1.unwrap();
+        if t1.total() == 0 {
+            return;
         }
+
+        std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
IMO, we should get the snapshot at https://github.com/tikv/tikv/pull/7524/files#diff-a8c2eaa50f7d51636324e25869027013R109 and reschedule the future instead of sleep here.